src/AutoRest.CSharp/Common/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -159,7 +157,8 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
                         }
                         else if (frameworkType == typeof(string) ||
                                  frameworkType == typeof(char) ||
-                                 frameworkType == typeof(Guid))
+                                 frameworkType == typeof(Guid) ||
+                                 ReferenceTypes.IsMgmtReferenceType(frameworkType))
```

Comment:
I wonder if we can extend the `JsonValueSerialization` with the concept of "SerializeAs" that would specify the resulting json type...

Reply:
yeah, was thinking the same thing, but given how large this change is already Id suggest we take that up in a later PR

Reply:
Please file a tracking issue and i'm fine w\ punting this bit.

Reply:
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5812